core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceMigrationActorSpec.scala

```diff
@@ -0,0 +1,731 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import cats.data.{NonEmptyList, OptionT, ReaderT}
+import cats.effect.IO
+import cats.effect.unsafe.IORuntime
+import cats.effect.unsafe.implicits.global
+import cats.implicits._
+import com.google.cloud.storage.{Acl, Storage}
+import com.google.longrunning.Operation
+import com.google.storagetransfer.v1.proto.TransferTypes.TransferJob
+import org.broadinstitute.dsde.rawls.dataaccess.slick.ReadWriteAction
+import org.broadinstitute.dsde.rawls.mock.{MockGoogleStorageService, MockGoogleStorageTransferService}
+import org.broadinstitute.dsde.rawls.model.{GoogleProjectId, GoogleProjectNumber, RawlsBillingAccountName, Workspace}
+import org.broadinstitute.dsde.rawls.monitor.migration.MigrationUtils.Outcome._
+import org.broadinstitute.dsde.rawls.monitor.migration.MigrationUtils.{Outcome, WorkspaceMigrationException}
+import org.broadinstitute.dsde.rawls.monitor.migration.WorkspaceMigration
+import org.broadinstitute.dsde.rawls.monitor.migration.WorkspaceMigrationActor._
+import org.broadinstitute.dsde.rawls.workspace.WorkspaceServiceSpec
+import org.broadinstitute.dsde.workbench.RetryConfig
+import org.broadinstitute.dsde.workbench.google2.GoogleStorageService
+import org.broadinstitute.dsde.workbench.google2.GoogleStorageTransferService.{JobName, JobTransferSchedule}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import org.broadinstitute.dsde.workbench.util2.{ConsoleLogger, LogLevel}
+import org.scalatest.Inspectors.forAll
+import org.scalatest.concurrent.Eventually
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{Assertion, BeforeAndAfterAll, OptionValues}
+import slick.jdbc.MySQLProfile.api._
+
+import java.sql.{SQLException, Timestamp}
+import java.util.UUID
+import scala.language.postfixOps
+
+class WorkspaceMigrationActorSpec
+  extends AnyFlatSpecLike
+    with BeforeAndAfterAll
+    with Matchers
+    with Eventually
+    with OptionValues {
+
+  val testKit: ActorTestKit = ActorTestKit()
+  implicit val logger = new ConsoleLogger("unit_test", LogLevel(false, false, true, true))
+  implicit val ec = IORuntime.global.compute
+  implicit val timestampOrdering = new Ordering[Timestamp] {
+    override def compare(x: Timestamp, y: Timestamp): Int = x.compareTo(y)
+  }
+
+  // This is a horrible hack to avoid refactoring the tangled mess in the WorkspaceServiceSpec.
+  val spec = new WorkspaceServiceSpec()
+
+
+  def runMigrationTest(test: MigrateAction[Assertion]): Assertion =
+    spec.withTestDataServices { services =>
+      test.run {
+        MigrationDeps(
+          services.slickDataSource,
+          GoogleProject("fake-google-project"),
```

Comment:
Before when we did this we read from a value in our test file. Could we do that here? I am always a little bit concerned when I see raw strings in test code that aren't parameterized, because it can lead to string proliferation across the tests (which becomes brittle)

Reply:
I'll move this into a variable and name it